src/codegen/expression.rs

```diff
@@ -3060,3 +3197,45 @@ fn code(loc: &Loc, contract_no: usize, ns: &Namespace, opt: &Options) -> Express
 
     Expression::AllocDynamicBytes(*loc, Type::DynamicBytes, size.into(), Some(code))
 }
+
+fn string_to_expr(
+    string: String,
+    cfg: &mut ControlFlowGraph,
+    ns: &Namespace,
+    contract_no: usize,
+    vartab: &mut Vartable,
+    opt: &Options,
+    func: Option<&Function>,
+) -> Expression {
+    let len_expr = ast::Expression::NumberLiteral {
+        loc: Loc::Codegen,
+        ty: Type::Uint(32),
+        value: BigInt::from(string.len()),
+    };
+
+    let expr = ast::Expression::AllocDynamicBytes {
```

Comment:
Since the string is known at compile time, can't we use `Expression::BytesLiteral` here?

Reply:
I think Bytes literal cannot be passed to a print instruction, as bytes literal is converted to an int and printed that way. Instead, it should be wrapped in a Expression::FormatString

Reply:
@seanyoung probably knows best

Reply:
`Expression::BytesLiteral()` is for `bytes4 foo = hex"41424344";`, not for strings. Emit will replace them with integers.